pkg/kubelet/kubelet.go

```diff
@@ -432,6 +433,10 @@ func (kl *Kubelet) Run(updates <-chan PodUpdate) {
 	if kl.dockerPuller == nil {
 		kl.dockerPuller = dockertools.NewDockerPuller(kl.dockerClient, kl.pullQPS, kl.pullBurst)
 	}
+	if kl.dockerCache == nil {
```

Comment:
We should do this in NewMainKubelet. It means some of the non Main initializations may need it. Those are all in tests.